pixeltable/catalog/table_version.py

```diff
@@ -383,6 +383,21 @@ def rename_column(self, old_name: str, new_name: str) -> None:
             self._update_md(ts, preceding_schema_version, conn)
         _logger.info(f'Renamed column {old_name} to {new_name} in table {self.name}, new version: {self.version}')
 
+    def update_attributes(self, num_retained_versions: Optional[int] = None, description: Optional[str] = None):
+        self.attributes = TableAttributes(
+            num_retained_versions = self.attributes.num_retained_versions if num_retained_versions is None else num_retained_versions,
+            description = self.attributes.description if description is None else description
+        )
+
+        # we're creating a new schema version
+        ts = time.time()
+        self.version += 1
+        preceding_schema_version = self.schema_version
+        self.schema_version = self.version
+        with Env.get().engine.begin() as conn:
+            self._update_md(ts, preceding_schema_version, conn)
+        _logger.info(f'Updated attributes: {self.attributes}, new version: {self.version}')
```

Comment:
Would be more interesting to log only the attributes that were changed, with the value.